src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -38,11 +37,111 @@
 #include <utils/mmeCauseUtils.h>
 #include "gtpCauseTypes.h"
 
+#include <locale>
+#include <memory.h>
+#include <signal.h>
+
+#include "epc/epctools.h"
+#include "epc/etevent.h"
+#include "epc/esocket.h"
+#include "epc/einternal.h"
+
+#include "epc/emgmt.h"
+#include "epc/etimerpool.h"
+
+#include "epc/epcdns.h"
+#include "epc/dnscache.h"
+#include "log.h"
+
+#define MCC_MNC_LEN 4
+#define LB_HB_LEN 8
+
 using namespace SM;
 using namespace mme;
 using namespace cmn;
 using namespace cmn::utils;
 
+
+cmn::ipc::IpcAddress destAddr_dns;
+struct CS_Q_msg cs_msg;
```

Comment:
Global variable addition is not good idea. It stops us making this code multi threaded later.